src/components/Comments/ReplyCommentItem.js

```diff
@@ -208,9 +224,27 @@ const ReplyCommentItem = ({
     },
     {
       id: 2,
-      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
-      icon: <SendDMAnonBlock />,
-      onPress: onPressDMAnon
+      name:
+        loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+          ? 'Loading...'
+          : 'Message using Incognito Mode',
+      icon: (
+        <IcDmAnon
+          color={
+            !userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+              ? COLORS.gray
+              : 'black'
+          }
+        />
+      ),
+      onPress: userAllowDm
+        ? onPressDMAnon
+        : () => {
+            Toast.show('This user does not allow messages in Incognito Mode.', Toast.SHORT);
+          },
+      style: (!userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon) && {
+        color: COLORS.gray300
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for displaying the option to send a message in incognito mode has been updated. Now it checks whether the user allows anonymous direct messages or not and displays the option accordingly. This is a necessary change for the new feature being implemented.
```diff
+ name:
+   loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+     ? 'Loading...'
+     : 'Message using Incognito Mode',
+ icon: (
+   <IcDmAnon
+     color={
+       !userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+         ? COLORS.gray
+         : 'black'
+     }
+   />
+ ),
+ onPress: userAllowDm
+   ? onPressDMAnon
+   : () => {
+       Toast.show('This user does not allow messages in Incognito Mode.', Toast.SHORT);
+     },
+ style: (!userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon) && {
+   color: COLORS.gray300
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->